consistency_check/src/bin/compressed_assets/main.rs

```diff
@@ -0,0 +1,431 @@
+                error!("Error writing failed proofs: {}", e);
+            }
+
+            if !shutdown_for_file_writer_rx.is_empty() {
+                break;
+            }
+
+            tokio::time::sleep(WRITER_SLEEP_TIME).await;
+        }
+
+        Ok(())
+    });
+
+    graceful_stop(&mut tasks).await;
+
+    shutdown_for_file_writer_tx.send(()).unwrap();
+
+    graceful_stop(&mut writers).await;
+
+    progress_bar.finish_with_message("Processing complete");
+}
+
+async fn verify_tree_batch(
+    progress_bar: Arc<ProgressBar>,
+    assets_processed: Arc<AtomicU64>,
+    assets_with_failed_proofs: Arc<AtomicU64>,
+    assets_with_missed_proofs: Arc<AtomicU64>,
+    rate: Arc<Mutex<f64>>,
+    shutdown_token: CancellationToken,
+    trees: Vec<String>,
+    rpc: Arc<RpcClient>,
+    rocks: Arc<Storage>,
+    failed_proofs: Arc<Mutex<HashMap<String, Vec<String>>>>,
+    failed_check: Arc<Mutex<HashSet<String>>>,
+    inner_workers: usize,
+) -> Result<(), JoinError> {
+    let api_metrics = Arc::new(metrics_utils::ApiMetricsConfig::new());
+
+    for tree in trees {
+        let t_key = match Pubkey::from_str(&tree) {
+            Ok(t) => t,
+            Err(e) => {
+                error!("Could not convert tree string into Pubkey: {}", e);
+                let mut f_ch = failed_check.lock().await;
+                f_ch.insert(tree.clone());
+                continue;
+            }
+        };
+
+        let tree_config_key = Pubkey::find_program_address(&[t_key.as_ref()], &mpl_bubblegum::ID).0;
+
+        if let Ok(tree_data) = rpc.get_account_data(&tree_config_key).await {
+            if let Ok(des_data) = mpl_bubblegum::accounts::TreeConfig::from_bytes(&tree_data) {
+                // spawn not more then N threads
+                let semaphore = Arc::new(Semaphore::new(inner_workers));
+
+                for asset_index in 0..des_data.num_minted {
+                    if shutdown_token.is_cancelled() {
+                        info!("Received shutdown signal");
+                        return Ok(());
+                    }
+
+                    let permit = semaphore.clone().acquire_owned().await.unwrap();
+
+                    let rocks_cloned = rocks.clone();
+                    let api_metrics_cloned = api_metrics.clone();
+                    let failed_proofs_cloned = failed_proofs.clone();
+                    let tree_cloned = tree.clone();
+                    let assets_processed_cloned = assets_processed.clone();
+                    let assets_with_failed_proofs_cloned = assets_with_failed_proofs.clone();
+                    let assets_with_missed_proofs_cloned = assets_with_missed_proofs.clone();
+                    let rate_cloned = rate.clone();
+                    let progress_bar_cloned = progress_bar.clone();
+                    tokio::spawn(async move {
+                        let asset_id = mpl_bubblegum::utils::get_asset_id(&t_key, asset_index);
```

Comment:
get_asset_id throw  panic!("Unable to find a viable program address bump seed")
I think we should process it here to avoid interruption of the entire verification process

Reply:
thats barely happen. because we generate asset ids with indexes which were already used in key generation, such as we take supply number from tree config first and then generate asset id for each index.